File: viewholderarrayadapter/src/main/java/com/commit451/viewholderarrayadapter/TextViewHolder.java
package com.commit451.viewholderarrayadapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

/**
 * Easy easy easy Text view view holder
 */
class TextViewHolder<T> extends RecyclerView.ViewHolder {

    public TextView text;

    TextViewHolder(View itemView) {
        super(itemView);
        text = (TextView) itemView.findViewById(android.R.id.text1);
    }

    public void bind(T object) {
        text.setText(object.toString());
    }
}

File: viewholderarrayadapter/src/main/java/com/commit451/viewholderarrayadapter/TextViewHolder.kt
package com.commit451.viewholderarrayadapter

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

/**
 * Easy easy easy Text view view holder holder
 */
class TextViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    var text: TextView? = itemView.findViewById(android.R.id.text1)

    fun bind(`object`: T) {
        text?.text = `object`.toString()
    }
}
